provider-ci/internal/pkg/templates/parameterized-go/Makefile

```
@@ -0,0 +1,262 @@
+	.pulumi/bin/pulumi plugin install #{{ or .Kind "resource" }}# #{{ .Name }}# #{{ .Version }}#
+	#{{- end }}#
+	@touch $@
+.PHONY: install_plugins
+
+lint_provider: provider
+	golangci-lint run -c .golangci.yml
+
+# `lint_provider.fix` is a utility target meant to be run manually
+# that will run the linter and fix errors when possible.
+lint_provider.fix:
+	golangci-lint run -c .golangci.yml --fix
+
+.PHONY: lint_provider lint_provider.fix
+
+#{{- if .Config.BuildProviderCmd }}#
+build_provider_cmd = #{{ if .Config.BuildProviderPre -}}#
+                         #{{ .Config.BuildProviderPre }}#;
+                     #{{- end -}}#
+                     #{{ .Config.BuildProviderCmd }}#
+#{{- else }}#
+build_provider_cmd = #{{ if .Config.BuildProviderPre -}}#
+                         #{{ .Config.BuildProviderPre }}#;
+                     #{{- end -}}#
+                     GOOS=$(1) GOARCH=$(2) CGO_ENABLED=0 go build $(PULUMI_PROVIDER_BUILD_PARALLELISM) -o "$(3)" -ldflags "$(LDFLAGS)" $(GO_MODULE)/cmd/$(PROVIDER)
+#{{- end }}#
+
+provider: bin/$(PROVIDER)
+
+# `make provider_no_deps` builds the provider binary directly, without ensuring that
+# `cmd/pulumi-resource-#{{ .Config.Provider }}#/schema.json` is valid and up to date.
+# To create a release ready binary, you should use `make provider`.
+provider_no_deps:
+	$(call build_provider_cmd,$(shell go env GOOS),$(shell go env GOARCH),$(WORKING_DIR)/bin/$(PROVIDER))
+bin/$(PROVIDER):
+	$(call build_provider_cmd,$(shell go env GOOS),$(shell go env GOARCH),$(WORKING_DIR)/bin/$(PROVIDER))
+.PHONY: provider provider_no_deps
+
+test:
+	cd #{{ .Config.TestFolder }}# && go test -parallel $(TESTPARALLELISM) -timeout 2h $(value GOTESTARGS)
+.PHONY: test
+
+#{{- if .Config.TestProviderCmd }}#
+test_provider_cmd = #{{ .Config.TestProviderCmd }}#
+#{{- else }}#
+test_provider_cmd = go test \
+	-coverprofile="coverage.txt" \
+	-coverpkg="./..." \
+	-parallel $(TESTPARALLELISM) \
+	./pkg/...
+#{{- end }}#
+test_provider:
+	$(call test_provider_cmd)
+.PHONY: test_provider
+
+# To make an immediately observable change to .ci-mgmt.yaml:
+#
+# - Edit .ci-mgmt.yaml
+# - Run make ci-mgmt to apply the change locally.
+#
+ci-mgmt: .ci-mgmt.yaml
+	go run github.com/pulumi/ci-mgmt/provider-ci@master generate
+.PHONY: ci-mgmt
+
+# Because some codegen depends on the version of the CLI used, we install a local CLI
+# version pinned to the same version as `go.mod`.
+#
+# This logic compares the version of .pulumi/bin/pulumi already installed. If it matches
+# the desired version, we just print. Otherwise we (re)install pulumi at the desired
+# version.
+.pulumi/bin/pulumi: .pulumi/version
+	@if [ -x .pulumi/bin/pulumi ] && [ "v$$(cat .pulumi/version)" = "$$(.pulumi/bin/pulumi version)" ]; then \
+		echo "pulumi/bin/pulumi version: v$$(cat .pulumi/version)"; \
+		touch $@; \
+	else \
+		curl -fsSL https://get.pulumi.com | \
+			HOME=$(WORKING_DIR) sh -s -- --version "$$(cat .pulumi/version)"; \
+	fi
```

This would be nice as a shared makefile include perhaps in `scripts/pulumi-cli.mk` (we can't use the `.mk` folder because of the existing `include $(wildcard .mk/*.mk)`).

```suggestion
include scripts/pulumi-cli.mk
```

---

This one ever so slightly different because go.mod location is different. It is a parameter though so perhaps I can add Go template expanders to customize it. I will try.

---

Yep I think this code can be shared as well.